library/src/scala/runtime/DynamicTuple.scala

```diff
@@ -169,7 +173,7 @@ object DynamicTuple {
     case _ =>
       xs match {
         case xs: TupleXXL => xs
-        case xs => TupleXXL(xs.productIterator.map(_.asInstanceOf[Object]).toArray)
+        case xs => TupleXXL.fromIArray(xs.productIterator.map(_.asInstanceOf[Object]).toArray.asInstanceOf[IArray[Object]])
```

Comment:
It's better to create IArray through its constructors instead of cast.

Reply:
Same as before, need `Iterator.toIArray`